app/models/piece.rb

```diff
@@ -23,18 +23,21 @@ def is_obstructed?(x_target, y_target)
 
 
   #determines if the move is horizontal
-  def horizontal_move?(x_target, y_target)
+  def horizontal_move?(x, y, x_target, y_target)
+    return if (x_target - x).abs > 0 && (y - y_target).abs ==  0
```

Comment:
Should this be `return true if` ? Or is the "true" implicit? (I'm not totally sure!) If it does need the "true" here, it'll have to be added to #vertical_move? and #diagonal_move? also.

Reply:
Since if statements always return booleans, the "true" is implicit. We could add it, but it would be unnecessary :)

Reply:
Thanks for the clarification! I didn't realize that about "if statements." I guess it makes sense though, since the last statement evaluated (the if statement, in this case) is always returned.